var/spack/repos/builtin/packages/py-contourpy/package.py

```diff
@@ -12,13 +12,20 @@ class PyContourpy(PythonPackage):
     homepage = "https://github.com/contourpy/contourpy"
     pypi = "contourpy/contourpy-1.0.5.tar.gz"
 
+    version("1.1.1", sha256="96ba37c2e24b7212a77da85004c38e7c4d155d3e72a45eeaf22c1f03f607e8ab")
     version("1.0.7", sha256="d8165a088d31798b59e91117d1f5fc3df8168d8b48c4acc10fc0df0d0bdbcc5e")
     version("1.0.5", sha256="896631cd40222aef3697e4e51177d14c3709fda49d30983269d584f034acc8a4")
 
     depends_on("python@3.8:", when="@1.0.7:", type=("build", "link", "run"))
     depends_on("python@3.7:", type=("build", "link", "run"))
-    depends_on("py-pybind11@2.6:", type=("build", "link"))
-    depends_on("py-setuptools@42:", type="build")
-    depends_on("py-numpy@1.16:", type=("build", "run"))
+    depends_on("meson@1.2:", when="@1.1.1:", type=("build"))
+    depends_on("py-meson-python@0.13.1:", when="@1.1:", type=("build"))
+    depends_on("py-pybind11@2.10.4:", when="@1.1:", type=("build", "link"))
+    depends_on("py-pybind11@2.6:", when="@:1.0", type=("build", "link"))
 
+    depends_on("py-numpy@1.26:1", when="^python@3.12:", type=("build", "run"))
+    depends_on("py-numpy@1.16:1", when="^python@:3.11", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-numpy@1.16:1", when="^python@:3.11", type=("build", "run"))
    depends_on("py-numpy@1.20:1", when="^python@:3.11", type=("build", "run"), when="@1.2")
```
In main the upper limit has already been removed, so I expect 1.3 to be depend on `py-numpy@1.20:` including allowing for py-numpy-2.